pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java

```diff
@@ -531,7 +547,7 @@ private InstancePartitions getInstancePartitions(TableConfig tableConfig,
             InstancePartitionsUtils.persistInstancePartitions(_helixManager.getHelixPropertyStore(),
                 instancePartitions);
           }
-          return instancePartitions;
+          return Pair.of(instancePartitions, true);
```

Comment:
We want to check if it is the same as existing instance partitions. If so, we can also skip the persisting step